interview-questions.md

```diff
@@ -7,60 +7,60 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 3. Ruby has an implicit return. What does that mean?
 
-  Your answer:
+  Your answer: I do believe an implicit return is a return that wasn't exactly asked for but Ruby has returned the last executable instructions. Meaning that Ruby doesn't really know what you're asking for but it will show you where it the code broke.
 
-  Researched answer:
+  Researched answer: So I was kind of right according to the syllabus it states that implicit return means the value of the last line of a method is automatically returned without using the keyword return. When using implicit returns, object literals must be wrapped in parenthesis so that the curly braces are not mistaken for the opening of the function’s body.
 
 
 
 4. What is object-oriented programming? How is it different than functional programming?
 
-  Your answer:
+  Your answer: Object oriented programming deals with building your project or code by using objects like building blocks and functional programming is like building the project or the code for functionality not structure wise.
 
-  Researched answer:
+  Researched answer:My research the definition of object oriented programming is s a programming paradigm that relies on the concept of classes and objects. It is used to structure a software program into simple, reusable pieces of code blueprints (usually called classes), which are used to create individual instances of objects. The definition of functional programming is s a programming paradigm that relies on the concept of classes and objects. It is used to structure a software program into simple, reusable pieces of code blueprints (usually called classes), which are used to create individual instances of objects. By reading these two definitions you can see that they are as different as No butter and jelly. They are good by themselves but they're really good when put together.
 
 
 
 5. What is the difference between a class and an object?
 
-  Your answer:
+  Your answer: A class is like a container and that holds objects and values in an object can be all different types of data. Classes are just a organizational tool for those objects.So to me the big difference between them is organization.
 
-  Researched answer:
+  Researched answer: During my research I found that a class is a template for objects. Objects are instance of a class. An object has a state in which all of its properties have values that you either explicitly define or that are defined by default settings. So like I was saying earlier classes are like blue prints of a building and the building materials are your objects. I just wish knowing all the methods and the santex was as easy to understand what the difference between a class and object is.
```

Comment:
A container and a blueprint are kind of different. Different enough that you might get some confusion if you talk about them interchangeably. Think of blueprints for a building, does it "contain" a building?  Rather it's a written concept of plans for which you can build any number of building that all share the same attributes.